### ISSUE TITLE
Adding loading from end of audio file

### ISSUE THREAD

**Issue by oulfik** [librosa/librosa#1434]
**Is your feature request related to a problem? Please describe.**
I already had an use case where I wanted to load an audio file from the end and not from the beginning. E.g. last 5 s of a file.  I think thats currently not possible because you have to know the audio length beforehand. 

**Describe the solution you'd like**
I changed the load function so that its possible to read from an end of a file. All you need to do is use a negative number for the offset parameter. Is this feature nice to have? If so, I could make a contribution.

```
from soundfile import SEEK_END
...

        with context as sf_desc:
            sr_native = sf_desc.samplerate
            if offset:
                if offset < 0:
                    # Seek to the end of the target read
                    sf_desc.seek(int(offset * sr_native), whence=SEEK_END)
                else:
                    # Seek to the start of the target read
                    sf_desc.seek(int(offset * sr_native))
            if duration is not None:
                frame_duration = int(duration * sr_native)
            else:
                frame_duration = -1
```

**Comment by bmcfee** [comment]
Interesting, I didn't realize that soundfile had this functionality already.  It does seem useful in certain circumstances.

I'm not 100% on adding it right now though, mainly because it wouldn't carry over well to the audioread backend.  I'd be fine with adding this when we remove audioread support (ie when soundfile gets mp3 support), but if we add before then it would require some rather ugly / duplicated code.

In the meantime, I think you can reasonably work around this by opening the soundfile object outside of load, seeking as you need, and then passing the soundfile object to `load` to handle the rest of the processing.

**Comment by xuboot** [comment]
url = "https://sneezoramafunction.blob.core.windows.net/usersounds/00bd9b8c914947d48724fd7e0c88041b.mp3"
    #url = "https://downsc.chinaz.net/Files/DownLoad/sound1/202111/14994.wav"
    x =io.BytesIO(urlopen(url).read())
    data, samplerate = sf.read(x)

Currently support mp3?
error :
        raise RuntimeError(prefix + _ffi.string(err_str).decode('utf-8', 'replace'))
RuntimeError: Error opening <_io.BytesIO object at 0x0000024A2111FC50>: File contains data in an unknown format.

Is there any way to solve it？

**Comment by bmcfee** [comment]
Bumping this to the 1.0 release.  The audioread deprecation is now expired (in the 1.0 dev branch), so this feature is unblocked.

**Comment by bmcfee** [comment]
I'm finally coding this up now, and seem to have bumped into a bug upstream from us.  The code as posted above more or less works on wav and mp3 files, but fails on ogg files in some pretty catastrophic ways.  The failures do not seem to be due to floating point rounding problems.  I'll try to work up a minimal example to push upstream later this week.

**Comment by bmcfee** [comment]
Indeed - this appears to be a known issue specifically with vorbis : https://github.com/libsndfile/libsndfile/issues/885

Opus works fine in my tests.

**Comment by bmcfee** [comment]
Fixed by #2030
